### ISSUE TITLE
Mismatched IP address?

### ISSUE THREAD

**Issue by ebiancarelli** [mill1000/midea-ac-py#48]
After a few weeks of running the integration, 3 of my devices went offline. Has anyone else seen the case where the device IPs differ between LAN and cloud?

➜  ~ msmart-ng discover
INFO:msmart.cli:Discovering all devices on local network.
WARNING:msmart.discover:Reported device IP 192.168.1.86 does not match received IP 192.168.1.85.
INFO:msmart.cloud:Using Midea cloud server: https://mp-prod.appsmb.com (China: False).
WARNING:msmart.discover:Reported device IP 192.168.1.51 does not match received IP 192.168.1.50.
WARNING:msmart.discover:Reported device IP 192.168.1.247 does not match received IP 192.168.1.245.
WARNING:msmart.cloud:Request to https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=/v1/user/login/id/get timed out.
WARNING:msmart.cloud:Request to https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=/v1/user/login/id/get timed out.
INFO:msmart.lan:Creating new connection to 192.168.1.86:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.51:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.249:6444.
INFO:msmart.lan:Authenticating with 192.168.1.249:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.246:6444.
INFO:msmart.lan:Authenticating with 192.168.1.246:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.247:6444.

**Comment by mill1000** [comment]
Hi,

I haven't seen this before. But glad I added the warning just in case. I'm guessing the discovery process failed to get information from the 3 devices with mismatched IPs?

> Has anyone else seen the case where the device IPs differ between LAN and cloud?

I don't think this has anything to do with the cloud. In fact it looks like your cloud connections failed. Do you have it blocked?

Anyway, the fix is probably to update the code to use the actual IP, not the IP the device thinks it has.

**Comment by ebiancarelli** [comment]
I've removed all the 'entities' for them and those 3 devices can no longer be added to the integration. 

I don't have the cloud api blocked. at least i don't think i do.

**Comment by mill1000** [comment]
> I've removed all the 'entities' for them and those 3 devices can no longer be added to the integration.

Do you receive an error? Did you remove just the entities or the integration entry too?

**Comment by ebiancarelli** [comment]
I just removed the non-functional entities. I kept the integration because there are several devices that still work properly.

When I 'auto' discover new devices, 3 show up but fail to be added.

**Comment by mill1000** [comment]
Can you try this branch and let me know if it fixes your problem?
```
pip install git+https://github.com/mill1000/midea-msmart@issue/ip_mismatch
```

**Comment by ebiancarelli** [comment]
```
$ msmart-ng -v
msmart version: 2023.10.1.dev1+gc6779f5
```

$ msmart-ng discover
INFO:msmart.cli:Discovering all devices on local network.
WARNING:msmart.discover:Reported device IP 192.168.1.86 does not match received IP 192.168.1.85. Using received IP.
INFO:msmart.cloud:Using Midea cloud server: https://mp-prod.appsmb.com (China: False).
WARNING:msmart.discover:Reported device IP 192.168.1.51 does not match received IP 192.168.1.50. Using received IP.
WARNING:msmart.discover:Reported device IP 192.168.1.247 does not match received IP 192.168.1.245. Using received IP.
INFO:msmart.lan:Creating new connection to 192.168.1.85:6444.
INFO:msmart.lan:Authenticating with 192.168.1.85:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.50:6444.
INFO:msmart.lan:Authenticating with 192.168.1.50:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.246:6444.
INFO:msmart.lan:Authenticating with 192.168.1.246:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.249:6444.
INFO:msmart.lan:Authenticating with 192.168.1.249:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.245:6444.
INFO:msmart.lan:Authenticating with 192.168.1.245:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.201:6444.
INFO:msmart.lan:Authenticating with 192.168.1.201:6444.
INFO:msmart.lan:Creating new connection to 192.168.1.64:6444.
INFO:msmart.lan:Authenticating with 192.168.1.64:6444.
INFO:msmart.lan:Authenticating with 192.168.1.85:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:43, Local key: 31e75e71785b2638b0ac5c1503e3c3a974de1151592da7ad699a14f8c1d4c194
INFO:msmart.lan:Authenticating with 192.168.1.50:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:44, Local key: c2c7a1a613c6ae2eb4043662ef4a571e5dd1a7c72e8fa52d1e8fa537ae1e9823
INFO:msmart.lan:Authenticating with 192.168.1.246:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:45, Local key: 3d57bc0d32cbe70817a51c868a620a94de51465a17a566276144a7f924b3a3cb
INFO:msmart.lan:Authenticating with 192.168.1.249:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:45, Local key: 8d30f51b9f67c069018967ebf478df332fde44358ef843633f0bb15c95089ac7
INFO:msmart.lan:Authenticating with 192.168.1.245:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:46, Local key: f6033641b58f9c0a3e1451e61fc0aeaacd1272dfe5701452c0ee144a4156aca2
INFO:msmart.lan:Authenticating with 192.168.1.201:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:47, Local key: 98b1c1372379a980ebf6c66459f73c0292c04ca0afbba778f8c570a4343aedd6
INFO:msmart.lan:Authenticating with 192.168.1.64:6444.
INFO:msmart.lan:Authentication successful. Expiration: 2023-11-01T11:11:47, Local key: 60edf6d6e6e688002bc7af26fe85ae3a39a53c6a78e58abbe8f4eb5ac13c5a3a
INFO:msmart.cli:Found 7 devices.
INFO:msmart.cli:Found device:
{'ip': '192.168.1.50', 'port': 6444, 'id': 148434069758851, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_242E', 'sn': '000000P0000000Q1AC93C4C2242E0000', 'key': '4c8ac8486bd247d795ebd89bb58939e4abafca3d65a748d6837949cced83f4d8', 'token': 'c981b350779c271de49d3ae9f4cbcc1d26562be27b1e96a33bbc0c218ee999d566bd899fe0f9681e27bb40113d5911decdb9c2708bc1b22d81d2b4643c5f231d'}
INFO:msmart.cli:Found device:
{'ip': '192.168.1.249', 'port': 6444, 'id': 147334558366101, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_49A0', 'sn': '000000P0000000Q1AC93C4C249A00000', 'key': 'bd61f802b20f4c70956f6bf26e42d32a69e3482ccfab4f7a905bbd45392f96de', 'token': '1a0848a9aded57ab8ded8624fe6692566d914a8418e73ac0c9199c2f1b36349fd3c356fa6498c4ef39071c46e645443f19603956455582aa756af48de38f99d1'}
INFO:msmart.cli:Found device:
{'ip': '192.168.1.245', 'port': 6444, 'id': 149533581388832, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_2ED8', 'sn': '000000P0000000Q1AC93C4C12ED80000', 'key': '611360951eb742efbdb28e070272f54bc46f293c23074db18cfe4da86a03f440', 'token': '6d3426ea67f8f374b30d3123ee5c006664a75b39b52185f6653f23d11c155bacecb14f3463343f700d50cf2dcb91e958df5bbde6df908d5b21d9de5969a7dfe0'}
INFO:msmart.cli:Found device:
{'ip': '192.168.1.201', 'port': 6444, 'id': 148434069757357, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_55BE', 'sn': '000000P0000000Q1AC93C42C55BE0000', 'key': 'd8b42f4f4d4846edb1d0ac09144bd66dc5f7a4cf73814f16a9ba99caa00384b8', 'token': '6535bdcba41214e784ef407bbb37f22a078c33d7538da4a7d3e1e14783d52a237b36ac27145b32ccd6d3e161a61df783d8c62deab9d8d4a5be426db2625228a8'}
INFO:msmart.cli:Found device:
{'ip': '192.168.1.85', 'port': 6444, 'id': 147334558362359, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_FDA8', 'sn': '000000P0000000Q1AC93C427FDA80000', 'key': 'aeb63f0822e54a6a8570bb41d488a2f81444750348ec40fda3e371a93340a5c4', 'token': '26a9335e3a238b52afb741250c08636316444e3264aed3871eac414b1db944d666b811fa086cba3bbf4debe5ae816c3c7228a539e6baff1958b6a23cf680b862'}
INFO:msmart.cli:Found device:
{'ip': '192.168.1.64', 'port': 6444, 'id': 149533581386631, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_FD92', 'sn': '000000P0000000Q1AC93C427FD920000', 'key': 'b0fa31a19c1f4f78bf6b675e3f5ee6450fd4731564ef45c78f6b2726b39e1647', 'token': 'b7dfe5edc0914ae62377412bdaa826761a6de41cd67bd8a4aa170ea9167708e93b915b9d2f07f135e44d45a77fc1ed6005c3bd6f982d1ce8806686b125d7e48e'}
INFO:msmart.cli:Found device:
{'ip': '192.168.1.246', 'port': 6444, 'id': 149533581400685, 'online': True, 'supported': True, 'type': <DeviceType.AIR_CONDITIONER: 172>, 'name': 'net_ac_499E', 'sn': '000000P0000000Q1AC93C4C2499E0000', 'key': '05ca9ac299c441d6ae63bb58faad2d4a277a61848aca41f9b8198027cfcc8415', 'token': 'b614931c824727ffe07877bb5335c55ea2215e2a53786d56b853e87dd513761ce50431b48a6bd9541aa35be5075a3655c0d3526b38769fbf090a1809f1c218ee'}

**Comment by mill1000** [comment]
Thanks! Looks good to me, all devices were found and are online. I'll merge the fix and get a release out soon.

**Comment by mill1000** [comment]
Fixed and released in 2023.11.0
